include/seqan3/alphabet/detail/debug_stream_alphabet.hpp

```diff
@@ -15,38 +15,56 @@
 
 namespace seqan3
 {
-/*!\name Formatted output overloads
- * \{
- */
-/*!\brief All alphabets can be printed to the seqan3::debug_stream by their char representation.
- * \tparam alphabet_t Type of the alphabet to be printed; must model seqan3::alphabet.
- * \param s The seqan3::debug_stream.
- * \param l The alphabet letter.
- * \relates seqan3::debug_stream_type
+
+/*!
+ * \brief The printer used for formatted output of seqan3::alphabet types.
+ *
+ * Prints the char representation of the given alphabet letter.
+ *
+ * \tparam alphabet_t The type of the alphabet to be printed.
+ * \ingroup alphabet
  */
-template <typename char_t, alphabet alphabet_t>
-inline debug_stream_type<char_t> & operator<<(debug_stream_type<char_t> & s, alphabet_t && l)
-    requires (!output_stream_over<std::basic_ostream<char_t>, alphabet_t>)
+template <typename alphabet_t>
+    requires alphabet<alphabet_t>
+struct alphabet_printer<alphabet_t>
```

Comment:
We don't have a style guide on whether to use short-hand notation or a `requires` for short stuff :D

Reply:
This PR touches a lot of code from the early development and some things didn't workout back then. I would favour the short notation as it is easier to read in my opinion.